src/cmake/GeosxOptions.cmake

```diff
@@ -108,7 +108,7 @@ message( "CMAKE_CXX_COMPILER_ID = ${CMAKE_CXX_COMPILER_ID}" )
 blt_append_custom_compiler_flag( FLAGS_VAR CMAKE_CXX_FLAGS DEFAULT "${OpenMP_CXX_FLAGS}" )
 blt_append_custom_compiler_flag( FLAGS_VAR CMAKE_CXX_FLAGS
                                  GNU   "-Wall -Wextra -Wpedantic -pedantic-errors -Wshadow -Wfloat-equal -Wcast-align -Wcast-qual"
-                                 CLANG "-Wall -Wextra -Wpedantic -pedantic-errors -Wshadow -Wfloat-equal -Wcast-align -Wcast-qual"
+                                 CLANG "-Wall -Wextra -Wpedantic -pedantic-errors -Wshadow -Wfloat-equal -Wno-cast-align -Wcast-qual"
```

Comment:
This warning seems a bit dangerous to disable, although it is kept on gcc at least. What was the problem with it, are we casting something to/from `localIndex *` that we shouldn't?

Reply:
We have to update python in the TPL's and then re-enable this. The error was:
```
/usr/gapps/GEOSX/thirdPartyLibs/python/linux-rhel7-x86_64-clang@10.0.0/python@3.8.5/include/python3.8/cpython/abstract.h:91:11: error: cast from 'char *' to 'vectorcallfunc *' (aka '_object *(**)(_object *, _object *const *, unsigned long, _object *)') increases required alignment from 1 to 8 [-Werror,-Wcast-align]
```

Reply:
IIUC this error comes from the python code in `cpython/abstract.h`?
Warnings from TPL includes should not pop if they are included between `<>` and/or with the `-isystem` from the command line (in `gcc`).
It looks like it's working with `GNU` (since you did not update the line). But the same behavior should occur with `clang`?